src/proto/world.proto

```diff
@@ -23,12 +23,13 @@ message Field
 
 message World
 {
-    required Timestamp time_sent  = 1;
-    required Field field          = 2;
-    required Team friendly_team   = 3;
-    required Team enemy_team      = 4;
-    required Ball ball            = 5;
-    required GameState game_state = 6;
+    required Timestamp time_sent    = 1;
+    required Field field            = 2;
+    required Team friendly_team     = 3;
+    required Team enemy_team        = 4;
+    required Ball ball              = 5;
+    required GameState game_state   = 6;
+    required uint64 sequence_number = 7;
```

Comment:
We should be careful adding new `required` fields to common protobuf messages as it will break backwards/forward compatibility (e.g. If a robot is running an older Thunderloop, it will receive some extra information for the sequence number field which it doesn't know how to parse (forward compatibility). An example of breaking backwards compatibility would be replaying old matches in Thunderscope, as Thunderscope won't be able to parse the old World messages anymore due to them not having the sequence number information).
A common practice to avoid breaking backwards compatibility would be to add new types as `optional`.
You can read more about this here: https://developers.google.com/protocol-buffers/docs/proto#updating

imo it's okay for us to add the new field as `required` and to break backwards compatibility since we don't have that much information that we will be losing, but I thought I'd point that out. 
@itsarune @LiCody What are your thoughts?

Reply:
Oh, I originally intended to add the sequence_number field as `optional`, but I'm not sure how I ended up adding it as `required`. I think it makes sense to change it to `optional`